compose/internal/service/record.go

```diff
@@ -184,6 +218,49 @@ func (svc record) Find(filter types.RecordFilter) (set types.RecordSet, f types.
 	return
 }
 
+func (svc record) Import(ses *RecordImportSession, ssvc ImportSessionService) error {
+	if ses.Decoder == nil {
+		return nil
+	}
+
+	if ses.Progress.StartedAt != nil {
+		return errors.New("Unable to start import: Import session already active")
+	}
+
+	sa := time.Now()
+	ses.Progress.StartedAt = &sa
+	ssvc.SetRecordByID(svc.ctx, ses.SessionID, 0, 0, nil, &ses.Progress, nil)
+
+	return svc.db.Transaction(func() (err error) {
+		err = (*ses.Decoder).Records(ses.Fields, func(mod *types.Record) error {
```

Comment:
@darh is something like this (`(*ses.Decoder).Records`) ok; it doesn't seem ok to me

Reply:
nope.